File: ruby/fizzbuzz.rb
# Fizz Buzz
#########
#  LeetCode problem number 412.
#  https://leetcode.com/problems/fizz-buzz/
#  Given an integer 'n', return a string array 'answer'(1-indexed) where:
#  - answer[i] == "FizzBuzz" if i is divisible by 3 and 5
#  - answer[i] == "Fizz" if i is divisible by 3
#  - answer[i] == "Buzz" if i is divisible by 5
#  - answer[i] == i (as a string) if none of the above conditions are true
#########
# Example
# -------
#  Input: n = 5
# Output: ["1","2","Fizz","4","Buzz"
##########

def fizz_buzz(n)
    answer = []
    
    n.times do |i|
        if (i + 1) % 5 == 0 && (i + 1) % 3 == 0
            answer << 'FizzBuzz'
        elsif (i + 1) % 5 == 0
            answer << 'Buzz'
        elsif (i + 1) % 3 == 0
            answer << 'Fizz'
        else 
            answer << "#{i + 1}"
        end
    end
    answer
end

File: python/fizzbuzz.py
def fizzbuzz(n: int):
    for num in range(n + 1):
        if num == 0:
            continue
        if num % 3 == 0 and num % 5 == 0:
            print('FizzBuzz')
        elif num % 3 == 0:
            print('Fizz')
        elif num % 5 == 0:
            print('Buzz')
        else:
            print(num)

fizzbuzz(15)

